src/lib/get-html.ts

```diff
@@ -1,14 +1,67 @@
+import marked from 'marked';
 import { Config } from './config';
-import { getMarked } from './get-marked-with-highlighter';
+import { getHighlightRenderer } from './get-marked-with-highlighter';
+import { getHeadingRenderer, TableOfContent, toc } from './get-marked-with-toc';
 
 /**
  * Generates a HTML document from a markdown string and returns it as a string.
  */
-export const getHtml = (md: string, config: Config) => `<!DOCTYPE html>
-<html>
-<head><meta charset="utf-8"></head>
-<body class="${config.body_class.join(' ')}">
-${getMarked(config.marked_options)(md)}
-</body>
-</html>
-`;
+export const getHtml = (md: string, config: Config) => {
+	marked.setOptions(config.marked_options);
+
+	(marked as any).use({ renderer: getHighlightRenderer(config.marked_options) });
+	(marked as any).use({ renderer: getHeadingRenderer(config).renderer });
+
+	const markedContent: string = marked(md);
+	let html = `<!DOCTYPE html>
+	<html>
+		<head><meta charset="utf-8"></head>
+		<body class="${config.body_class.join(' ')}">
+			${markedContent}
+		</body>
+	</html>
+	`;
+
+	// generate table of content only if we have headings
+	// const { toc } = getHeadingRenderer(config);
+	if (html.includes('<!-- TOC -->') && toc.length !== 0) {
+		const ctx = [];
+		ctx.push(`<div id="table-of-contents"><h1>${config.toc_headings ?? 'Table of content'}</h1>\n<ul>`);
```

Comment:
You're already setting a default value for `toc_heading` so you don't need to repeat a fallback value here.

Also, if `config.toc_heading = ''`, it should omit the `h1` alltogether.

BTW not sure whether the TOC heading should be level 1 or level 2 🤷🏻‍♂️

Reply:
I have set toc_skip to 0 as default. This way the users will not be in situation where it don't know why it's title is not display. I have update the reade.me to describe this skip functionnality